pkg/resources/common/formatter.go

```diff
@@ -78,21 +78,35 @@ func formatter(summarycache *summarycache.SummaryCache) types.Formatter {
 		if err != nil {
 			return
 		}
+		userInfo, ok := request.GetUserInfo()
+		if !ok {
+			return
+		}
+		accessSet := asl.AccessFor(userInfo)
+		if accessSet == nil {
+			return
+		}
+		hasUpdate := accessSet.Grants("update", gvr.GroupResource(), resource.APIObject.Namespace(), resource.APIObject.Name())
+		hasDelete := accessSet.Grants("delete", gvr.GroupResource(), resource.APIObject.Namespace(), resource.APIObject.Name())
+
 		selfLink := selfLink(gvr, meta)
 
 		u := request.URLBuilder.RelativeToRoot(selfLink)
 		resource.Links["view"] = u
 
-		if _, ok := resource.Links["update"]; !ok && slice.ContainsString(resource.Schema.CollectionMethods, "PUT") {
-			resource.Links["update"] = u
```

Comment:
Previously we would set `resource.Links["update"] = u` when it `resource.Links["update"]` didn't exist. We're now assuming it should be explicitly set. While I do also prefer that style of only giving what is explicitly stated, is it possible that this may cause a regression? If a caller of the `DefaultTemplate` assumes the update portion will get filled in, after this change it won't. So in the UI example, maybe the "Edit" button is greyed out when it should be enabled.

Reply:
I think that the underlying access control system would have set this value to something, so I'm not too concerned about the value not being there when it should be. 

That being said, I think you are right that this change could cause a regressions since `u` may (and I think is) different than what is previously set. I'll look at adding a new statement that overrides the update link in the case where we are allowed and not blocked. Good Catch.

Reply:
I've made a change here to get this a bit closer to the old side. I think this still differs a bit from the old side (i.e. now if we know there's an update we add the link even if the underlying code never set it), but I think that this should do what we want. Please feel free to re-open if this doesn't fit with your thinking.